Client/Application/LeanplumIntegration.swift

```diff
@@ -0,0 +1,201 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import AdSupport
+import Shared
+import Leanplum
+
+private let LeanplumEnvironmentKey = "LeanplumEnvironment"
+private let LeanplumAppIdKey = "LeanplumAppId"
+private let LeanplumKeyKey = "LeanplumKey"
+
+private let log = Logger.browserLogger
+
+private enum LeanplumEnvironment: String {
+    case development = "development"
+    case production = "production"
+}
+
+enum LeanplumEventName: String {
+    case firstRun = "First Run"
+    case secondRun = "Second Run"
+    case openedApp = "Opened App"
+    case openedLogins = "Opened Login Manager"
+    case openedBookmark = "Opened Bookmark"
+    case openedNewTab = "Opened New Tab"
+    case interactWithURLBar = "Interact With Search URL Area"
+    case savedBookmark = "Saved Bookmark"
+    case openedTelephoneLink = "Opened Telephone Link"
+    case openedMailtoLink = "Opened Mailto Link"
+    case downloadedImage = "Download Media - Saved Image"
+    case savedLoginAndPassword = "Saved Login and Password"
```

Comment:
From Leanplum, about a month back:
"We do not have a hard limit for event names, but they can become difficult to read and work with when they get very long. 
I tested with this event name and had no issues: 
adfssdfsdfadfssdfsdfadfssdfsdfadfssdfsdfadfssdfsdfadfssdfsdfadfssdfsdfadfssdfsdfadfssdfsdfadfssdfsdfadfssdfsdfadfssdfsdfadfssdfsdfadfssdfsdf"

I am using standard english here as the audience for consuming them is the marketing team.